lib/versionmanager.js

```diff
@@ -328,6 +328,26 @@ function getCurrentDependencies(pkgData = {}, options = {}) {
   return allDependencies
 }
 
+async function getIgnoredUpgrades(current, upgraded, upgradedPeerDependencies, options = {}) {
+  const [upgradedLatestVersions, latestVersions] = await upgradePackageDefinitions(
+    current,
+    { ...options, peer: false, peerDependencies: undefined, loglevel: 'silent' }
+  )
```

Comment:
Doesn't `analyzeProjectDependencies` already call `upgradePackageDefinitions`? This would result in doubling all requests to `queryVersions` and `getPeerDependencies`. That seems like a poor design unless I'm missing something.

Reply:
Yes and no.

This is necessary, since the target functions (e.g. `latest`) directly filter out all other versions than the one desired. There is no chance to access the latest compatible version (with option `peerDependencies`) and the latest incompatible version (without option `peerDependecies`) at the same time. `analyzeProjectDependencies` calls `upgradePackageDefinitions` with peer dependencies and this code calls it without.

However, the requests aren't doubled since you cache all responses already:
https://github.com/raineorshine/npm-check-updates/blob/v11.5.4/lib/package-managers/npm.js#L151-L154

Therefore, the second call is executed from memory and hence runs very fast.

Btw., `getPeerDependencies` isn't called twice.

Reply:
Excellent points! Thanks so much for walking me through that. Some day I would like to re-architect npm-check-updates to have a cleaner pipeline with distinct fetch, label, and filter phases. In the mean time, this will work fine.

I forgot about the caching! That was added for `--deep` but it's really handy here.